wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -262,6 +266,31 @@ abstract class Wallet
     sendToOutputs(destinations, feeRate, fromAccount, reserveUtxos)
   }
 
+  override def makeOpReturnCommitment(
+      message: String,
+      hashMessage: Boolean,
+      feeRate: FeeUnit,
+      fromAccount: AccountDb): Future[Transaction] = {
+    val messageToUse = if (hashMessage) {
+      CryptoUtil.sha256(ByteVector(message.getBytes)).bytes
+    } else {
+      if (message.length > 32) {
+        throw new IllegalArgumentException(
+          s"Message cannot be greater than 32 characters, it should be hashed, got $message")
+      } else ByteVector(message.getBytes)
+    }
+
+    val asm = Seq(OP_RETURN,
+                  BytesToPushOntoStack(messageToUse.size),
```

Comment:
this won't work for small things (think 1 bit), you should use the calculate push op method in script util